src/resourcemanager/resourcemanager.cpp

```diff
@@ -0,0 +1,479 @@
+
+        object.set("name", device.mName.CStr());
+        object.set("sharedCount", device.mSharedCount);
+        object.set("groups", ToJSONArray(device.mGroups));
+        object.set("hostDevices", ToJSONArray(device.mHostDevices));
+
+        array.add(object);
+    }
+
+    return array;
+}
+
+static Poco::JSON::Array DumpMounts(const Array<FileSystemMount>& mounts)
+{
+    Poco::JSON::Array array;
+
+    for (const auto& mount : mounts) {
+        Poco::JSON::Object object;
+
+        object.set("destination", mount.mDestination.CStr());
+        object.set("type", mount.mType.CStr());
+        object.set("source", mount.mSource.CStr());
+        object.set("options", ToJSONArray(mount.mOptions));
+
+        array.add(object);
+    }
+
+    return array;
+}
+
+static Poco::JSON::Array DumpHosts(const Array<Host>& hosts)
+{
+    Poco::JSON::Array array;
+
+    for (const auto& host : hosts) {
+        Poco::JSON::Object object;
+
+        object.set("ip", host.mIP.CStr());
+        object.set("hostName", host.mHostname.CStr());
+
+        array.add(object);
+    }
+
+    return array;
+}
+
+static Poco::JSON::Array DumpResources(const Array<ResourceInfo>& resources)
+{
+    Poco::JSON::Array array;
+
+    for (const auto& resource : resources) {
+        Poco::JSON::Object object;
+
+        object.set("name", resource.mName.CStr());
+        object.set("groups", ToJSONArray(resource.mGroups));
+        object.set("mounts", DumpMounts(resource.mMounts));
+        object.set("env", ToJSONArray(resource.mEnv));
+        object.set("hosts", DumpHosts(resource.mHosts));
+
+        array.add(object);
+    }
+
+    return array;
+}
+
+/***********************************************************************************************************************
+ * JSONProvider
+ **********************************************************************************************************************/
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+Error JSONProvider::DumpNodeConfig(const sm::resourcemanager::NodeConfig& config, String& json) const
+{
+    try {
+        Poco::JSON::Object object;
+
+        object.set("version", config.mVersion.CStr());
+        object.set("nodeType", config.mNodeConfig.mNodeType.CStr());
+        object.set("priority", config.mNodeConfig.mPriority);
+        object.set("devices", DumpDevices(config.mNodeConfig.mDevices));
+        object.set("resources", DumpResources(config.mNodeConfig.mResources));
+        object.set("labels", ToJSONArray(config.mNodeConfig.mLabels));
+
+        std::ostringstream ss;
+        Poco::JSON::Stringifier::stringify(object, ss);
+
+        json = ss.str().c_str();
+    } catch (const Poco::Exception& e) {
+        LOG_ERR() << "Error dumping config: " << e.what();
```

Comment:
No need to duplicate error log. Error should be displayed in place where it is checked.